linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/IncOptimizer.scala

```diff
@@ -741,6 +769,36 @@ final class IncOptimizer private[optimizer] (config: CommonPhaseConfig, collOps:
       jsConstructorDef.map(_.optimizedDef)
   }
 
+  private final class JSTopLevelMethodContainer extends JSMethodContainer {
+
+    private[this] var methods = Map.empty[(String, String), (JSMethodImpl, Position)]
+
+    val untrackedJSClassCaptures: List[ParamDef] = Nil
+    def untrackedThisType(namespace: MemberNamespace): Type = NoType
+
+    def updateWith(topLevelExports: List[LinkedTopLevelExport]): Unit = {
+      val newMethods = topLevelExports.map(_.tree).collect {
+        case m: TopLevelMethodExportDef =>
+          val key = (m.moduleID, m.topLevelExportName)
+          val impl = methods.get(key).fold(new JSMethodImpl(this, key))(_._1)
+          impl.updateWith(m.methodDef)
+          key -> (impl, m.pos)
+      }.toMap
+
+      methods
+        .withFilter(e => !newMethods.contains(e._1))
+        .foreach(_._2._1.delete())
+
+      methods = newMethods
+    }
+
+    def optimizedMethod(moduleID: String, name: String): TopLevelMethodExportDef = {
+      val (impl, pos) = methods((moduleID, name))
+      val newMethod = impl.optimizedDef.asInstanceOf[JSMethodDef]
+      TopLevelMethodExportDef(moduleID, newMethod)(pos)
```

Comment:
Consider using `newMethod.pos` here, so that you don't have to keep track of positions in `methods`.

Or else store the `TopLevelMethodExportDef` as values of `methods` instead.

Reply:
> Consider using newMethod.pos here,

Wouldn't that move the position form the annotation (that triggered the export) to the method definition?

> Or else store the TopLevelMethodExportDef as values of methods instead.

I'm not sure I understand. You mean as second element of the tuple (where pos is right now)? Or make a separate JSTopLevelMethodImpl (that's not so easy, because `Processable` requires a `version` member).

Reply:
Ah yes indeed. Both points make sense.

Let's leave it as is, then.